src/test/java/dev/openfeature/sdk/ThreadLocalTransactionContextPropagatorTest.java

```diff
@@ -0,0 +1,57 @@
+package dev.openfeature.sdk;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.FutureTask;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ThreadLocalTransactionContextPropagatorTest {
+
+    ThreadLocalTransactionContextPropagator contextPropagator = new ThreadLocalTransactionContextPropagator();
+
+    @Test
+    public void setTransactionContextOneThread() {
+        EvaluationContext firstContext = new ImmutableContext();
+        contextPropagator.setTransactionContext(firstContext);
+        assertSame(firstContext, contextPropagator.getTransactionContext());
+        EvaluationContext secondContext = new ImmutableContext();
+        contextPropagator.setTransactionContext(secondContext);
+        assertNotSame(firstContext, contextPropagator.getTransactionContext());
+        assertSame(secondContext, contextPropagator.getTransactionContext());
+    }
+
+    @Test
+    public void emptyTransactionContext() {
+        EvaluationContext result = contextPropagator.getTransactionContext();
+        assertNull(result);
+    }
+
+    @SneakyThrows
+    @Test
+    public void setTransactionContextTwoThreads() {
+        EvaluationContext firstContext = new ImmutableContext();
+        EvaluationContext secondContext = new ImmutableContext();
+
+        Callable<EvaluationContext> callable = () -> {
+            assertNull(contextPropagator.getTransactionContext());
+            contextPropagator.setTransactionContext(secondContext);
+            EvaluationContext transactionContext = contextPropagator.getTransactionContext();
+            assertSame(secondContext, transactionContext);
+            return transactionContext;
+        };
+        contextPropagator.setTransactionContext(firstContext);
+        EvaluationContext firstThreadContext = contextPropagator.getTransactionContext();
+        assertSame(firstContext, firstThreadContext);
+
+        FutureTask<EvaluationContext> futureTask = new FutureTask<>(callable);
+        Thread thread = new Thread(futureTask);
+        thread.start();
+        EvaluationContext secondThreadContext = futureTask.get();
+
+        assertSame(secondContext, secondThreadContext);
+        assertSame(firstContext, firstThreadContext);
```

Comment:
Here we should also get and assert the value of the `firstThreadContext` after line 52.
Currently as far as I can tell the test does not verify that `firstThreadContext` is not changed by the callable in the second thread, as the value of `firstThreadContext` is read before (line 46) the second thread is started (line 51)

Reply:
Agreed.